src/core/processing/qgsprocessingparameters.cpp

```diff
@@ -5585,3 +5629,104 @@ void QgsProcessingParameterLayoutItem::setItemType( int type )
 {
   mItemType = type;
 }
+
+//
+// QgsProcessingParameterColor
+//
+
+QgsProcessingParameterColor::QgsProcessingParameterColor( const QString &name, const QString &description, const QVariant &defaultValue, bool optional )
+  : QgsProcessingParameterDefinition( name, description, defaultValue, optional )
+{
+
+}
+
+QgsProcessingParameterDefinition *QgsProcessingParameterColor::clone() const
+{
+  return new QgsProcessingParameterColor( *this );
+}
+
+QString QgsProcessingParameterColor::valueAsPythonString( const QVariant &value, QgsProcessingContext & ) const
+{
+  if ( !value.isValid() || value.isNull() )
+    return QStringLiteral( "None" );
+
+  if ( value.canConvert<QgsProperty>() )
+    return QStringLiteral( "QgsProperty.fromExpression('%1')" ).arg( value.value< QgsProperty >().asExpression() );
+
+  if ( value.canConvert< QColor >() && !value.value< QColor >().isValid() )
+    return QStringLiteral( "QColor()" );
+
+  if ( value.canConvert< QColor >() )
+    return QStringLiteral( "'%1'" ).arg( value.value< QColor >().name() );
```

Comment:
Maybe better to export in the ARGB format to avoid loosing alpha value or make it depended on the alpha value presense?

Reply:
I've tweaked that now -- now it uses a QColor object value instead.